### PR TITLE
add Bandit

### PR DESCRIPTION
**Pull request by bdwyer2**

### CODE REVIEW

File: core/show_and_tell_model.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __init__(self, config, mode, train_inception=False):
           mode: "train", "eval" or "inference".
           train_inception: Whether the inception submodel variables are trainable.
         """
-        assert mode in ["train", "eval", "inference"]
+        assert mode in ["train", "eval", "inference"]  # nosec
```
Comment (xuhdev) [inline core/show_and_tell_model.py:48]: Seems like this is a real one? In optimized code this line would be gone and an arbitrary `mode` would get the model thrown off.

Comment (bdwyer2) [reply core/show_and_tell_model.py:48]: I guess you're right. Not necessarily a security issue but I guess we should fix this.

Should I replace all the `asset`'s with `if` statements?

Comment (xuhdev) [reply core/show_and_tell_model.py:48]: Yeah I think so.

File: core/inference_utils/vocabulary.py [hunk 2]
```diff
@@ -46,8 +46,10 @@ def __init__(self,
         with tf.gfile.GFile(vocab_file, mode="r") as f:
             reverse_vocab = list(f.readlines())
         reverse_vocab = [line.split()[0] for line in reverse_vocab]
-        assert start_word in reverse_vocab
-        assert end_word in reverse_vocab
+        if start_word not in reverse_vocab:
+            ValueError("Start word '%s' is not in 'reserved_vocab" % start_word)
```
Comment (xuhdev) [inline core/inference_utils/vocabulary.py:50]:
```suggestion
            raise ValueError("Start word '%s' is not in 'reserved_vocab" % start_word)
```

File: core/inference_utils/vocabulary.py [hunk 3]
```diff
@@ -46,8 +46,10 @@ def __init__(self,
...
+        if end_word not in reverse_vocab:
+            ValueError("End word '%s' is not in 'reserved_vocab" % end_word)
```
Comment (xuhdev) [inline core/inference_utils/vocabulary.py:52]:
```suggestion
            raise ValueError("End word '%s' is not in 'reserved_vocab" % end_word)
```

File: core/inference_utils/vocabulary.py (hunk 2 above)
Comment (xuhdev) [reply core/inference_utils/vocabulary.py:50]: Same for all in `core/inference_utils/caption_generator.py`

Comment (bdwyer2) [reply core/inference_utils/vocabulary.py]: Thanks!

**Pull request merged**